spec/helpers.lua

```diff
@@ -1473,13 +1944,13 @@ end
 -- Modified version of `pl.utils.executeex()` so the output can directly be
 -- used on an assertion.
 -- @name execute
--- @param cmd command to execute
+-- @param cmd command string to execute
 -- @param pl_returns (optional) boolean: if true, this function will
 -- return the same values as Penlight's executeex.
--- @return if pl_returns is true, returns four return values
--- (ok, code, stdout, stderr); if pl_returns is false,
+-- @return if `pl_returns` is true, returns four return values
+-- (ok, code, stdout, stderr); if `pl_returns` is false,
 -- returns either (false, stderr) or (true, stderr, stdout).
-local function exec(cmd, pl_returns)
+function exec(cmd, pl_returns)
```

Comment:
This seems to be creating a global variable called `exec`? Is it on purpose?

```suggestion
local function exec(cmd, pl_returns)
```

Reply:
no, there is a forward declaration here: https://github.com/Kong/kong/blob/a606c2a0aa74f862ad71dc3622bfada3f564c3f5/spec/helpers.lua#L830

there are some refrences that would otherwise fail, hence the forward declaration